app/src/org/commcare/dalvik/dialogs/AlertDialogFactory.java

```diff
@@ -19,6 +19,8 @@
 
     private final AlertDialog dialog;
     private final View view;
+    private DialogInterface.OnCancelListener cancelListener;
```

Comment:
Wish I could find a way to do this that didn't involve saving the cancel listener and manually calling it.